.github/workflows/gpuci.yml

```diff
@@ -0,0 +1,41 @@
+name: "GPU CI/CD"
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - 'r**'
+    types: [ labeled, synchronize ]
+
+jobs:
+  gpu-test:
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'gpuci' || github.event_name == 'synchronize' }}
+
+    steps:
+    - name: Run nvidia-smi test
+      run: |
+        whoami
+        nvidia-smi
+
+    - name: Checkout the PR code
+      uses: actions/checkout@v3
+
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
+
+    - name: Install NeMo-Curator and PyTest
+      run: |
+        pip install wheel cython
+        RAPIDS_NIGHTLY=1 pip install --extra-index-url=https://pypi.anaconda.org/rapidsai-wheels-nightly/simple ".[cuda12x]"
```

Comment:
I think it makes sense to use RAPIDS Nightly Builds here, but please feel free to comment if not? I just took this command straight from our README.

Alternatively, we could just do `pip install --extra-index-url https://pypi.nvidia.com ".[cuda12x]"`.

Also, maybe it would make sense to directly point to [requirements_rapids_nightly.txt](https://github.com/NVIDIA/NeMo-Curator/blob/main/requirements/requirements_rapids_nightly.txt)/[requirements_cuda12x.txt](https://github.com/NVIDIA/NeMo-Curator/blob/main/requirements/requirements_cuda12x.txt) and [requirements.txt](https://github.com/NVIDIA/NeMo-Curator/blob/main/requirements/requirements.txt) instead?

Reply:
I also don't know about including `--no-cache-dir`?

Reply:
`--no-cache-dir` is optional, it's useful on machines with limited storage, but not a big issue either way.

Curious on others thoughts here, but I prefer running GPU tests without Rapids nightly. I see pytests on pr's as a way to check if we broke any gpu functionality. 
As a followup ( I can assist if needed) we can set up a nightly or weekly job that runs these same pytests with Rapids nightlies to identify any upstream breakages.

Reply:
Thanks! Left a comment about this in https://github.com/NVIDIA/NeMo-Curator/issues/259 as a reminder.